packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -298,15 +312,39 @@ class _CodeViewState extends State<CodeView>
                         Expanded(
                           child: LayoutBuilder(
                             builder: (context, constraints) {
-                              return Lines(
-                                constraints: constraints,
-                                scrollController: textController,
-                                lines: lines,
-                                pausedFrame: pausedFrame,
-                                searchMatchesNotifier:
-                                    widget.controller.searchMatches,
-                                activeSearchMatchNotifier:
-                                    widget.controller.activeSearchMatch,
+                              final num longestLineLength = lines.fold(
+                                  0,
+                                  (longestLength, line) => math.max(
+                                      longestLength,
+                                      line.toPlainText().length));
+                              final fileWidth =
+                                  CodeView.assumedCodeCharacterWidth *
+                                      longestLineLength;
+                              final boxWidth = math.max(
+                                  constraints.minWidth, constraints.maxWidth);
+
+                              return Scrollbar(
+                                isAlwaysShown: fileWidth > boxWidth,
+                                controller: horizontalController,
+                                child: SingleChildScrollView(
+                                  scrollDirection: Axis.horizontal,
+                                  controller: horizontalController,
+                                  child: Lines(
+                                    constraints: BoxConstraints(
```

Comment:
Nit: in Flutter the general rule is to prefer composition over inheritance.
In this case what that means is rather than writing
```dart
Lines(constraints: foo, ...)
```
where the Lines build method adds a ConstrainedBox, write
```dart
ConstrainedBox(constraints: foo, child: Lines(...))
```
If there was a reason the Lines widget needed to know about the constraints then that would be different and passing in the constraints would be fine.

Reply:
In addition, I suspect you do not need to specify any constraint for the Y axis and that for the X axis you can clarify things by making the constraint more fixed. I may be getting things mixed up but I believe you should be able to just write
SizedBox(width: fileWidth)

Reply:
`Lines` was already being passed `constraints` (it looks like to figure out where to scroll to when searching the file: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/debugger/codeview.dart#L572-L593). However all it uses is the `maxHeight`, so I could refactor to use `SizedBox` and only pass the height to `Lines`. Wdyt?

Reply:
you are right. fine to leave as constraints or switch to SizedBox to be clearer.

Reply:
Switched to `SizedBox`, and now passing only the height to `Lines`